src/moonlink/src/storage/iceberg/rest_catalog.rs

```diff
@@ -0,0 +1,272 @@
+    }
+
+    async fn table_exists(&self, table: &TableIdent) -> IcebergResult<bool> {
+        self.catalog.table_exists(table).await
+    }
+
+    async fn rename_table(&self, _src: &TableIdent, _dest: &TableIdent) -> IcebergResult<()> {
+        todo!("rename table is not supported");
+    }
+
+    async fn update_table(&self, mut _commit: TableCommit) -> IcebergResult<Table> {
+        todo!("update table is not supported");
+    }
+
+    async fn register_table(
+        &self,
+        __table: &TableIdent,
+        _metadata_location: String,
+    ) -> IcebergResult<Table> {
+        todo!("register existing table is not supported")
+    }
+}
+
+#[cfg(feature = "storage-gcs")]
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use iceberg::spec::{NestedField, PrimitiveType, Schema, Type};
+    use reqwest::Client;
+    const REST_CATALOG_PROP_URI: &str = "uri";
+
+    async fn ensure_namespace(catalog: &IcebergRestCatalog, ns: &str) {
+        let ns_ident = NamespaceIdent::from_vec(vec![ns.to_string()]).unwrap();
+        let _ = catalog.create_namespace(&ns_ident, HashMap::new()).await;
+    }
+
+    #[tokio::test]
+    async fn test_create_table() {
+        let builder = IcebergRestCatalogBuilder::default().with_client(Client::new());
+        let catalog = builder
+            .load(
+                "test",
+                HashMap::from([(
+                    REST_CATALOG_PROP_URI.to_string(),
+                    "http://localhost:8181".to_string(),
+                )]),
+            )
+            .await;
+
+        let binding = catalog.unwrap();
+        ensure_namespace(&binding, "ns1").await;
+
+        let builder = IcebergRestCatalogBuilder::default().with_client(Client::new());
+        let catalog = builder
+            .load(
+                "test",
+                HashMap::from([(
+                    REST_CATALOG_PROP_URI.to_string(),
+                    "http://localhost:8181".to_string(),
+                )]),
+            )
+            .await;
+        assert!(catalog.is_ok());
+        let table_creation = TableCreation::builder()
+            .name("test1".to_string())
+            .schema(
+                Schema::builder()
+                    .with_fields(vec![
+                        NestedField::optional(1, "foo", Type::Primitive(PrimitiveType::String))
+                            .into(),
+                        NestedField::required(2, "bar", Type::Primitive(PrimitiveType::Int)).into(),
+                        NestedField::optional(3, "baz", Type::Primitive(PrimitiveType::Boolean))
+                            .into(),
+                    ])
+                    .build()
+                    .unwrap(),
+            )
+            .build();
+
+        let namespace = NamespaceIdent::new("ns1".to_string());
+        let result = catalog
+            .unwrap()
+            .create_table(&namespace, table_creation)
+            .await;
+        match result {
```

Comment:
I'm not sure if we ignore creation errors what's the point of the unit test..
I would suggest:
- Use a random string as table name
- Write a RAII-style class for table drop